pkg/spdk/replica.go

```diff
@@ -2083,7 +2083,32 @@ func (r *Replica) waitForRestoreComplete() error {
 	return nil
 }
 
-func (r *Replica) postIncrementalRestoreOperations(restore *Restore) error {
+func (r *Replica) postIncrementalRestoreOperations(spdkClient *spdkclient.Client, restore *Restore) error {
+	r.log.Infof("Replacing snapshot %v of the restored volume", restore.SnapshotName)
+
+	if r.restore.State == btypes.ProgressStateCanceled {
+		r.log.Info("Doing nothing for canceled backup restoration")
+		return nil
+	}
+
+	r.log.Infof("Deleting snapshot %v for snapshot replacement of the restored volume", restore.SnapshotName)
+	_, err := r.SnapshotDelete(spdkClient, restore.SnapshotName)
```

Comment:
What if this is the first snapshot of a DR volume?

Reply:
AFAIK if we delete the first snapshot it will get coalesced into the head. Any concerns?

Reply:
Oh YES. SPDK can coalesce snapshot content into the current head lvol. Plz add a comment for this.

Reply:
Added comment, thank you.